_data/harnesses/aggregating-average/kstreams.yml

```diff
@@ -0,0 +1,171 @@
+        - action: execute
+          file: tutorial-steps/dev/gradle-wrapper.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-gradle-wrapper.adoc
+
+        - action: execute
+          file: tutorial-steps/dev/make-configuration-dir.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-config-dir.adoc
+
+        - action: make_file
+          file: configuration/dev.properties
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-dev-file.adoc
+            
+    - title: Create a schema for the model obect
+      content:
+        - action: execute
+          file: tutorial-steps/dev/make-avro-dir.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-avro-dir.adoc
+
+        - action: make_file
+          file: src/main/avro/rating.avsc
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-avro-schema-rating.adoc
+        
+        - action: make_file
+          file: src/main/avro/countsum.avsc
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-avro-schema-countsum.adoc
+
+        - action: execute
+          file: tutorial-steps/dev/build-project.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/build-project.adoc
+
+    - title: Create the Kafka Streams topology
+      content:
+        - action: execute
+          file: tutorial-steps/dev/make-src-dir.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-src-dir.adoc
+            
+        - action: make_file
+          file: src/main/java/io/confluent/developer/RunningAverage.java
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/make-topology.adoc 
+
+    - title: Compile and run the Kafka Streams program
+      content:
+        - action: execute
+          file: tutorial-steps/dev/build-uberjar.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/build-uberjar.adoc
+
+        - action: execute_async
+          file: tutorial-steps/dev/run-dev-app.sh
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/run-dev-app.adoc
+
+    - title: Produce sample data to the input topic
+      content:
+        - action: execute
+          file: tutorial-steps/dev/console-producer.sh
+          stdin: tutorial-steps/dev/input-ratings.json
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/run-producer.adoc
+
+    - title: Consume data from the output topic
+      content:
+        - action: execute_async
+          file: tutorial-steps/dev/console-consumer.sh
+          stdout: tutorial-steps/dev/outputs/actual-outputs.json
+          render:
+            file: tutorials/aggregating-average/kstreams/markup/dev/run-consumer.adoc
```

Comment:
`run-consumer.adoc` states
> Before you start producing ratings, it’s a good idea to set up the consumer on the output topic.

But the consume step is currently after the produce step above.

Reply:
hm, makes total sense. it may be the reason that I'm getting an empty file with consumer results.